src/lib.rs

```diff
@@ -110,6 +110,18 @@ impl TlsAcceptor {
     }
 }
 
+#[async_tls_acceptor::async_trait]
```

Comment:
Since `TlsAcceptor::accept` returns a named future, this is more inefficient than using `TlsAcceptor::accept` directly. I haven't tried to see if that's possible, but if the trait can provide an API something like the following, it will be efficient enough. (Since `Output` is already a different type, it is not a problem that the type of `Future` is different between implementations.)

```rust
impl<Input> async_tls_acceptor::Acceptor<Input> for TlsAcceptor
where
    Input: AsyncRead + AsyncWrite + Send + Sync + Unpin + 'static,
{
    type Output = server::TlsStream<Input>;
    type Error = std::io::Error;
    type Future = Accept<Input>;
    fn accept(&self, input: Input) -> Self::Future {
        TlsAcceptor::accept(&self, input)
    }
}
```

Reply:
I believe that this is not possible on stable currently, as it requires GATs in order to express futures like this (for situations in which there is no named future, and the unnamed future is not static):

```rust
impl<Input> async_tls_acceptor::Acceptor<Input> for TlsAcceptor
where
    Input: AsyncRead + AsyncWrite + Send + Sync + Unpin + 'static,
{
    type Output = TlsStream<Input>;
    type Error = crate::Error;
    type Future<'a> = Pin<Box<dyn Future<Output = Result<Self::Output, Self::Error>> + Send + 'a>>;

    fn accept<'a>(&'a self, input: Input) -> Self::Future<'a> {
        Box::pin(TlsAcceptor::accept(&self, input))
    }
}
```

Please correct me if I'm mistaken. Boxing the future doesn't seem like a terribly onerous price to pay until async traits land

Reply:
You can work around that by making the lifetime a parameter of the trait itself. Hint: https://ytrizja.de/blog/2020/1001_rustiit4r.html